### ISSUE TITLE
TypeScript Client Emitter Incorrectly Generates Browser Code with Direct Module Imports and Missing `getClient` Implementation

### ISSUE THREAD

**Issue by mario-guerra** [Azure/autorest.typescript#2974]
The TypeSpec TypeScript client code emitter generates browser-specific output (located in the `dist/browser` directory) that incorrectly includes direct imports from the `@typespec/ts-http-runtime` module, rather than bundling or replacing these imports with browser-compatible code. 

**Detailed Description:**

When generating a TypeScript client using TypeSpec, the emitter produces a `dist/browser` directory intended for browser usage. However, the generated JavaScript files within this directory still contain direct imports from the `@typespec/ts-http-runtime` module, such as:

```javascript
import { getClient } from "@typespec/ts-http-runtime";
```

This is incorrect because:

*   **Browser Environments:** Browsers cannot directly load Node.js modules like `@typespec/ts-http-runtime`.
*   **`dist/browser` Purpose:** The `dist/browser` output is intended to be a self-contained bundle that does not rely on external Node.js modules at runtime.
*   **Incorrect Bundling:** The emitter should either bundle the relevant parts of `@typespec/ts-http-runtime` directly into the `dist/browser` output or replace the import with a reference to a browser-specific implementation of `getClient` that's already included in the bundle.

**Steps to Reproduce:**

1.  Create a TypeSpec project with a service definition that uses HTTP operations.
2.  Use the TypeSpec TypeScript client emitter to generate code.
3.  Examine the generated code in the `dist/browser` directory, for example anything that uses a client to interact with an API.
4.  Observe the direct imports from `@typespec/ts-http-runtime`.

**Environment:**

*   TypeSpec version: 0.63
*   TypeScript client emitter version: 0.37.0
*   Operating System: Windows 11
*   Node.js version: 18.19.0

**Additional Information:**

*   The `package.json` for `@typespec/ts-http-runtime` correctly specifies a browser entry point in the `"browser"` field.
*   The `tshy` build process is configured to generate browser-specific output, but the generated code still contains the direct imports and lacks the `getClient` implementation.
*   There is no apparent configuration option in `tspconfig.yaml` to specify that a browser client is desired.

**Comment by mario-guerra** [comment]
@allenjzhang for awareness

**Comment by qiaozha** [comment]
It sounds like a tshy bundle issue instead of codegen issue @mpodwysocki might have some thoughts here?

**Comment by MaryGao** [comment]
@mario-guerra 

**TL;DR**

We will not bundle all codes into one file. To use our generated library in the browser, you need to use a bundler. For details on how to do this, please refer to our [bundling documentation](https://aka.ms/AzureSDKBundling).

The long story is we export our browser version code with [node conditional exports](https://nodejs.org/api/packages.html#conditional-exports) under folder `dist/browser` in package.json. Here the `browser` should be a common known package condition and is expected to be recoginized with popular bounders like [webpack](https://webpack.js.org/guides/package-exports/#target-environment). Actually we mention this in our `README.md`([link](https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-test/test/schemaRegistry/generated/typespec-ts/README.md?plain=1#L31-L32)) but it seems in unbranded client we didn't populate this information in README file.

```
"exports": {
    "./package.json": "./package.json",
    ".": {
      "browser": {
        "types": "./dist/browser/index.d.ts",
        "default": "./dist/browser/index.js"
      },
}
```

You could also find the sample project [here](https://github.com/Azure/azure-sdk-for-js/tree/main/samples/Bundling), but let us know if any issue you meet!

**Comment by mario-guerra** [comment]
@MaryGao thanks for the link. I tried creating a bundle for the generated client code using webpack first, then rollup and I was unsuccessful with either bundler due to numerous errors. 

Has anyone been able to successfully create a bundle for unbranded client code generated from TypeSpec?

**Comment by MaryGao** [comment]
We've offline tried with both webpack and rollup yet and met issue with rollup tool: https://github.com/Azure/azure-sdk-for-js/issues/32430. Not sure the issue is similar or not?

**Comment by MaryGao** [comment]
@mario-guerra I think the issue should be resolved and let me know if any concern!
